cardano-testnet/src/Testnet/Property/Run.hs

```diff
@@ -60,17 +64,29 @@ runTestnet tn = do
       IO.exitFailure
 
 
-testnetProperty :: (Conf -> H.Integration ()) -> H.Property
-testnetProperty tn = integrationWorkspace "testnet" $ \workspaceDir -> do
-  conf <- mkConf workspaceDir
-
-  -- Fork a thread to keep alive indefinitely any resources allocated by testnet.
-  void . H.evalM . liftResourceT . resourceForkIO . forever . liftIO $ IO.threadDelay 10000000
-
-  void $ tn conf
-
-  H.failure -- Intentional failure to force failure report
-
+testnetProperty :: CardanoTestnetOptions -> (Conf -> H.Integration ()) -> H.Property
+testnetProperty CardanoTestnetOptions{cardanoOutputDir} tn =
+  case cardanoOutputDir of
```

Comment:
It may be more readable if the `case` would return `Conf` and there was a single `go` call. I'd suggest also renaming `go` to something more meaningful e.g. `forkAndRunTestnet`.

Reply:
I renamed `go` to `forkAndRunTestnet` :+1: 

> It may be more readable if the case would return Conf and there was a single go call

@carbolymer> I'm not sure how you can simplify this much, because the first call to `go` is nested within the call to `integrationWorkspace` and I'm not sure how to unify the two cases because of that.